lib/graphql/input_object_type.rb

```diff
@@ -96,7 +96,8 @@ def coerce_non_null_input(value, ctx)
         end
       end
 
-      GraphQL::Query::Arguments.new(input_values, argument_definitions: arguments)
+      arguments_class ||= GraphQL::Query::Arguments.construct_arguments_class(argument_definitions: arguments)
```

Comment:
I need this for a single test. I'm not sure if this is a better approach than defining than generating the classess during traversal of the schema. Thoughts?

Reply:
I don't think we should create a new class here, we should be sure to do it all ahead of time. Which test fails? I'd be happy to take a try at another approach.